NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -859,18 +837,33 @@ main(int argc, char **argv)
 	double timeInit = 0.0;
 	timerInit.startTimer();
 
-	std::string modelFile;
-	std::string hash;
+	std::shared_ptr<NFIQ::ModelInfo> modelInfoObj {};
 
 	try {
-		std::tie(modelFile, hash) = NFIQ2UI::parseModel(arguments);
+		modelInfoObj = NFIQ2UI::parseModel(arguments);
 	} catch (const NFIQ2UI::Exception &e) {
 		std::cerr << "Unable to extract model information. " << e.what()
 			  << "\n";
 		return EXIT_FAILURE;
 	}
 
-	const NFIQ::NFIQ2Algorithm model(modelFile, hash);
+	logger->debugMsg("Model Name: " + modelInfoObj->getModelName());
```

Comment:
```suggestion
	logger->debugMsg("Model Name: " + (modelInfoObj->getModelName().empty() ? "<NA>" : modelInfoObj->getModelName());
```
and so on. Suggestion for optional properties. Blank also works (and works as-is). See what looks better.